specs/agents/tracing-distributed-tracing.md

```diff
@@ -55,10 +114,55 @@ Agents MUST implement these validation rules when mutating `tracestate`:
 - Vendor values may only contain ASCII RFC0020 characters (i.e., the range `0x20` to `0x7E`) except comma `,` and `=`.
 - In addition to the above limitations, the keys and values used in the `es` entry must not contain the characters `:` and `;`.
-- If adding another key/value pair to the `es` entry would exceed the limit of 256 chars,
-  that key/value pair MUST be ignored by agents.
-  The key/value and entry separators `:` and `;` have to be considered as well.
+- If adding another key/value pair to the `es` entry would exceed the limit of
+  256 chars (including separator characters `:` and `;`), that key/value pair
+  MUST be ignored by agents.
 
 Note that we will currently only ever populate an `es` `tracestate` entry at the trace root.
 It is not strictly necessary to validate `tracestate` in its entirety when received downstream.
 Instead, downstream agents may opt to only parse the `es` entry and skip validation of other vendors' entries.
+This means that the vendor key validations are only relevant if an agent adds
+its own non-`es` keys to tracestate
+
+In addition, we do not enforce the 32-entry limit for vendor entries in
+`tracestate`. Doing so would cripple our ability to use `tracestate` for our
+own purposes, arbitrarily. Removing other entries to make way for our own
+would also cause unexpected behavior. In any case, this situation should be
+rare and we feel comfortable ignoring the validation rules in this case.
+
+
+### HTTP Headers
+
+Every outgoing request should be intercepted and modified to include both the
+`traceparent` and `tracestate` headers, described above.
+
+If an incoming request contains either of the `traceparent` or `tracestate`
+headers, they should be propagated throughout the transaction and mutated as
+specified above before being set on outgoing requests.
+
+The `span-id` part of the `traceparent` header should be the `id` of the span
+representing the outgoing request. If (and only if) that span is not sampled,
+the `span-id` may instead be the `id` of the current transaction.
+
+
+### Binary Fields
+
+Our implementation relies on the [W3C Binary Trace
+Context](https://w3c.github.io/trace-context-binary/) standard.  In order to
+make sure we are fully aligned, all agents are implementing the specification described in
+[this commit](https://github.com/w3c/trace-context-binary/blob/571cafae56360d99c1f233e7df7d0009b44201fe/spec/20-binary-format.md).
+
+Binary fields should only be used where strings are not allowed, such as in
+Kafka record headers. The field names should still be `traceparent` and
+`tracestate`.
+
+
+### Legacy HTTP Headers/Binary Fields
+
+Some agents support the legacy header name `elastic-apm-traceparent` and the
+binary field name `elasticapmtraceparent`. These names were used while the W3C
+standard was being finalized, to avoid any backwards-compatibility issues. New
+agents do not need to support these legacy names. Because `tracestate` was
```

Comment:
The [Java agent](https://github.com/elastic/apm-agent-java/blob/2259495d153766fe36aa2833a0455394b2fadbb8/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceContext.java#L89) still uses **only** `elasticapmtraceparent` for the binary format (it may be the only agent that implemented it so far).
As far as I can see, the [W3C spec](https://w3c.github.io/trace-context-binary/) is still a draft, so I think we can stick to this name until it becomes final.